tests/Doctrine/Tests/DBAL/ConnectionTest.php

```diff
@@ -403,6 +392,80 @@ public function testUpdateWithSameColumnInDataAndIdentifiers()
         );
     }
 
+    /**
+     * @group DBAL-2688
+     */
+    public function testUpdateWithIsNull()
+    {
+        $conn = $this->getExecuteUpdateMockConnection();
+
+        $conn->expects($this->once())
+            ->method('executeUpdate')
+            ->with(
+                'UPDATE TestTable SET text = ?, is_edited = ? WHERE id IS NULL AND name = ?',
+                [
+                    'some text',
+                    null,
+                    'foo',
+                ],
+                [
+                    'string',
+                    'null',
+                    'string',
+                ]
+            );
+
+        $conn->update(
+            'TestTable',
+            [
+                'text' => 'some text',
+                'is_edited' => null,
+            ],
+            [
+                'id' => null,
+                'name' => 'foo',
+            ],
+            [
+                'text' => 'string',
+                'is_edited' => 'null',
+                'id' => 'null',
```

Comment:
Specifying `'null'` type is a bit weird because Doctrine does not have such type. As the testsuite somewhat also serves as API usage documentation I would suggest not promoting misusage here. I am pretty sure that code won't work when actually executed as Doctrine should complain about not knowing `null` type.
IMO the best would be to not specify any type for `null` values and let Doctrine decide what to do with it. As an alternative, if you want to complete type specification here, you should use the approriate types that reflect the column type like for `id` you would probably pass `integer` as type.